src/components/views/ProductPage/ProductPageContainer.js

```diff
@@ -0,0 +1,16 @@
+import { connect } from 'react-redux';
+import ProductPage from './ProductPage';
+import { getProductById } from '../../../redux/productsRedux';
+
+const mapStateToProps = (state, props) => {
+  const id = props.match.params.id;
+  const filteredProducts = state.products.filter(product => product.id === id);
+  const productParams = filteredProducts[0] || {};
+
+  return {
+    ...productParams,
+    columns: getProductById(state, id),
```

Comment:
W `productsRedux` zdefiniowałaś funkcję `getProductByI` któa to szuka nam prouktu o wskazanym **id**, a w tym momencie wównież tutaj masz analogiczny kod do szukania produktu. 

Nie rozumiem też tej właściwości `columns`, czemu miała by służyć?